tests/test_tuning_XXXMBE.cpp

```diff
@@ -0,0 +1,157 @@
+				std::ifstream ifs(file_name);
+				std::string line, colname;
+
+				std::getline(ifs, line);
+				std::istringstream iss(line);
+				while (iss >> colname) {
+						column_names.push_back(colname);
+						raw_columns.emplace_back(colname);
+						columns[colname] = &(raw_columns.back());
+				}
+				const int num_columns = column_names.size();
+
+				// Read remaining lines into std::string vectors
+				int lineno = 1;
+				while (std::getline(ifs, line)) {
+						iss.str(line); iss.clear();
+						int i=0;
+						while (iss >> colname && i < num_columns) {
+								raw_columns[i].push_back(colname);
+								++i;
+						}
+						if (i > num_columns) {
+								std::cout << "Warning: Ignoring extra column(s) on line " << lineno << std::endl;
+						}
+						++lineno;
+				}
+		}
+
+		// Get data vectors of different types
+		std::vector<double> get_dvector(const std::string& colname) { return columns[colname]->dvalues(); }
+		std::vector<int> get_ivector(const std::string& colname) { return columns[colname]->ivalues(); }
+		// Default is to return double values
+		std::vector<double> operator[](const std::string& colname) { return columns[colname]->dvalues(); }
+		
+		std::vector<std::string> column_names;
+		std::vector<Column> raw_columns;
+		std::map<std::string, Column*> columns;
+};
+
+BOOST_AUTO_TEST_CASE(CheckMassBalanceWithinXXXMBE)
+{
+		//std::string case_name(boost::unit_test::framework::master_test_suite().argv[1]);
+		std::string case_name("01_TUNING_XXXMBE");
+		std::string file_name = case_name + ".INFOITER";
+
+		ColumnData data(file_name);
+		auto rstep = data.get_ivector("ReportStep");
+		auto tstep = data.get_ivector("TimeStep");
+		auto mbo = data["MB_Oil"];
+		auto mbw = data["MB_Water"];
+		auto mbg = data["MB_Gas"];
+
+		const int num_reports = 1 + *std::max_element(rstep.begin(), rstep.end());
+		std::vector<double> max_mb;
+		max_mb.reserve(num_reports);
+
+
+		// Find the maximum mass balance error at each converged time step for each report step.. 
+		const int nrows = rstep.size();
+		int rcur = 0;
+		int tcur = 0;
+		double max_mb_step = std::numeric_limits<double>::min();
+		for (int i=0; i<(nrows-1); ++i) {
+				if (tcur != tstep[i+1] || rcur != rstep[i+1]) {
+						max_mb_step = std::max({mbo[i], mbw[i], mbg[i], max_mb_step});
+						tcur = tstep[i+1];
+				}
+				if (rcur != rstep[i+1]) {
+						max_mb.push_back(max_mb_step);
+						max_mb_step = std::numeric_limits<double>::min();
+						rcur = rstep[i+1];
+				}			
+		}
+		max_mb.push_back( std::max({mbo.back(), mbw.back(), mbg.back(), max_mb_step}) );
+
+		std::cout << "---------------------------------------------------------------------------" << std::endl;
+		std::cout << "Found the following converged max mass balance error (per report step):" << std::endl;
+		for (auto& val : max_mb)
+				std::cout << val << std::endl;
+		std::cout << "---------------------------------------------------------------------------" << std::endl;
```

Comment:
Could you use [`BOOST_TEST_MESSAGE()`](https://www.boost.org/doc/libs/1_82_0/libs/test/doc/html/boost_test/test_output/test_tools_support_for_logging/test_output_macro_message.html) to output these messages instead?  The one drawback to `BOOST_TEST_MESSAGE()` is that you don't get the messages unless you pass
```
--log_level=message
```
to the test runner, but you **do** get the advantage of the thread-safe logger built into Boost.Test.